### PR TITLE
spec: Drawer component

### PR DESCRIPTION
**Pull request by stephenjwatkins**
## 📝 Changes

- create spec for Drawer component

### CODE REVIEW

File: documentation/specs/Drawer.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+   */
+  children: ReactNode;
+};
+
+type DrawerStandaloneContentAreaProps = {
+  /**
+   * Content area for a drawer without a banner (standalone).
+   */
+  children: ReactNode;
+};
```
Comment (stephenjwatkins) [inline documentation/specs/Drawer.md:116]: open to suggestions on naming and concepts.

the idea is to provide some built-ins for certain drawer content areas.

`Drawer.BanneredContentArea`:

image

`Drawer.StandaloneContentArea`:

image

this is admittedly a bit wordy but it allows for any flexibility for future `Drawer` designs. an alternative might be exposing a general `Drawer.ContentArea` with `padding` props.

Comment (OskiTheCoder) [reply documentation/specs/Drawer.md:116]: im good with the the naming you chose, its sensible to me

**Review by OskiTheCoder — APPROVED** [review]
